R/save.r

```diff
@@ -58,6 +60,29 @@ ggsave <- function(filename, plot = last_plot(),
   invisible()
 }
 
+#' Parse a DPI input from the user
+#'
+#' Allows handling of special strings when user specifies a DPI like "print".
+#'
+#' @param dpi Input value from user
+#' @return Parsed DPI input value
+#' @noRd
+parse_dpi <- function(dpi) {
+  if (is.character(dpi) && length(dpi) == 1) {
+    dpi <- switch(dpi,
+      screen = 72,
+      print = 300,
+      retina = 320,
+      stop("Unknown DPI string", call. = FALSE)
+    )
+    return(dpi)
+  } else if (is.numeric(dpi) && length(dpi) == 1) {
+    return(dpi)
```

Comment:
Done. I know assignment still silently returns a value, but left `dpi` as the last expression in the first `if` block to avoid confusion. Just looking at some neighbouring code this appears to be the style. Could be something to [add to the style guide](http://style.tidyverse.org/functions.html#return)?